### PR TITLE
[Test generation] Add DEPENDS argument

### PR DESCRIPTION
**Pull request by ProGTX**
1. Added `DEPENDS` argument to `generate_cts_test`
2. Added files for `math_builtin_api` test to depend on

### CODE REVIEW

File: tests/math_builtin_api/CMakeLists.txt [hunk 1]
```diff
@@ -5,6 +5,13 @@ set(MATH_VARIANT base double half)
 
 set(MATH_CAT integer geometric native half)
 
+set(math_builtin_depends
+  "math_builtin.h"
```
Comment (bader) [inline tests/math_builtin_api/CMakeLists.txt:9]: Should we put a header file as a dependency for source generation?

**Review by bader — APPROVED** [review]
LGTM, although I'm not sure if `math_builtin.h` should be in the list.

File: tests/math_builtin_api/CMakeLists.txt (hunk 1 above)
Comment (ProGTX) [reply tests/math_builtin_api/CMakeLists.txt]: I think you're right, it shouldn't have an impact on the generator

**Review by bader — APPROVED** [review]

**Comment by bader** [comment]
@keryell, could you take a look, please?

**Review by keryell — APPROVED** [review]
That looks good, from my limited CMake knowledge.

**Pull request merged**